backend/todo/adapter/mongo.py

```diff
@@ -0,0 +1,78 @@
+from pymongo import MongoClient, ReturnDocument
+
+from todo.port.models import Todo, Task
+from todo.port.database import Database
+
+import uuid
+
+from datetime import datetime
+
+
+class Mongo(Database):
+    def __init__(self) -> None:
+        client = MongoClient("mongodb://localhost")
+        db = client["local"]
+        self.collection = db["todo"]
+
+    def create(self, title: str, color: str, task: list[Task] | None) -> Todo:
+        """Create and return a todo with the given data."""
+        if task is None:
+            task = []
+
+        todo = Todo(
+            id=str(uuid.uuid4()),
+            created_at=datetime.now(),
+            title=title,
+            color=color,
+            task=task,
+        )
+
+        todo_data = todo.dict()
+        self.collection.insert_one(todo_data)
+        return todo_data
+
+    def list(self) -> list:
+        """Get and return a list of Todos in the system."""
+        todos = list(self.collection.find())
+        for todo in todos:
+            if todo:
+                todo["_id"] = str(todo["_id"])
+        return todos
+
+    def filtered_by(self, filter_by:str) -> list:
+        """Return list of Todos that comply with the received filter."""
+        if filter_by.lower() == "completed":
+            todos = list(self.collection.find({"task.completed": True}))
+            for todo in todos:
+                if todo:
+                    todo["_id"] = str(todo["_id"])
+            return todos
+        
+        if filter_by.lower() == "uncompleted":
+            todos = list(self.collection.find({"task.completed": False}))
+            for todo in todos:
+                if todo:
+                    todo["_id"] = str(todo["_id"])
+            return todos
+
+    def get(self, id: str) -> Todo:
+        """Get and return a specific Todo with the given ID."""
+        todo = self.collection.find_one({"id": id})
+
+        if todo:
+            todo["_id"] = str(todo["_id"])
+        return todo
```

Comment:
You are returning a Todo or None, which is not part of the Database protocol. If you can not get the Todo from the database, you should produce an Exception